stl/inc/flat_set

```diff
@@ -0,0 +1,718 @@
+        const iterator _Begin = begin();
+        const iterator _End   = end();
+
+        if (_Begin == _End) {
+            return;
+        }
+
+        _Sort_potentially_sorted(_Begin, _End);
+        _STL_INTERNAL_CHECK(_STD is_sorted(begin(), end(), _Get_comp()));
+
+        _Erase_dupes_if_needed();
+    }
+
+    _NODISCARD const _Container& _Get_cont() const noexcept {
+        return _My_pair._Get_first();
+    }
+
+    _NODISCARD _Container& _Get_cont() noexcept {
+        return _My_pair._Get_first();
+    }
+
+    _NODISCARD const key_compare& _Get_comp() const noexcept {
+        return _My_pair._Myval2;
+    }
+
+    _NODISCARD key_compare& _Get_comp() noexcept {
+        return _My_pair._Myval2;
+    }
+
+    _Compressed_pair<container_type, key_compare> _My_pair;
+};
+
+_EXPORT_STD struct sorted_unique_t {
+    explicit sorted_unique_t() = default;
+};
+_EXPORT_STD inline constexpr sorted_unique_t sorted_unique{};
+
+_EXPORT_STD struct sorted_equivalent_t {
+    explicit sorted_equivalent_t() = default;
+};
+_EXPORT_STD inline constexpr sorted_equivalent_t sorted_equivalent{};
+
+template <class _Kty, class _Keylt = less<_Kty>, class _Container = vector<_Kty>>
+_EXPORT_STD class flat_set
+    : public _Base_flat_set<_Kty, _Keylt, _Container, false, flat_set<_Kty, _Keylt, _Container>, sorted_unique_t> {
+public:
+    using _Base_flat_set<_Kty, _Keylt, _Container, false, flat_set, sorted_unique_t>::_Base_flat_set;
+};
+
+template <class _Kty, class _Keylt = less<_Kty>, class _Container = vector<_Kty>>
+_EXPORT_STD class flat_multiset : public _Base_flat_set<_Kty, _Keylt, _Container, true,
+                                      flat_multiset<_Kty, _Keylt, _Container>, sorted_equivalent_t> {
+public:
+    using _Base_flat_set<_Kty, _Keylt, _Container, true, flat_multiset, sorted_equivalent_t>::_Base_flat_set;
+};
+
+_EXPORT_STD template <class _Kty, class _Keylt, class _Container, class _Pred>
+size_t erase_if(flat_set<_Kty, _Keylt, _Container>& _Val, _Pred _Predicate) {
+    _TRY_BEGIN
+    return _Erase_remove_if(_Val, _Pass_fn(_Predicate));
+    _CATCH_ALL
+    _Val.clear();
+    _RERAISE;
+    _CATCH_END
+}
+
+_EXPORT_STD template <class _Kty, class _Keylt, class _Container, class _Pred>
+size_t erase_if(flat_multiset<_Kty, _Keylt, _Container>& _Val, _Pred _Predicate) {
+    _TRY_BEGIN
+    return _Erase_remove_if(_Val, _Pass_fn(_Predicate));
+    _CATCH_ALL
+    _Val.clear();
+    _RERAISE;
+    _CATCH_END
+}
+
+template <class _Kty, class _Keylt, class _Container, class _Alloc>
+struct uses_allocator<flat_set<_Kty, _Keylt, _Container>, _Alloc>
+    : bool_constant<uses_allocator_v<_Container, _Alloc>> {};
+
+template <class _Kty, class _Keylt, class _Container, class _Alloc>
+struct uses_allocator<flat_multiset<_Kty, _Keylt, _Container>, _Alloc>
+    : bool_constant<uses_allocator_v<_Container, _Alloc>> {};
+
+_EXPORT_STD template <class _Container, class _Keylt = less<typename _Container::value_type>>

```

Comment:
Deduction guides don't need `_EXPORT_STD`. They are already reachable and thus work in modules.